pkg/authorization/registry/clusterpolicy/registry.go

```diff
@@ -0,0 +1,114 @@
+package clusterpolicy
+
+import (
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/rest"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/watch"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	"github.com/openshift/origin/pkg/authorization/registry/policy"
+)
+
+// Registry is an interface for things that know how to store ClusterPolicies.
+type Registry interface {
+	// ListClusterPolicies obtains list of policies that match a selector.
+	ListClusterPolicies(ctx kapi.Context, label labels.Selector, field fields.Selector) (*authorizationapi.ClusterPolicyList, error)
+	// GetClusterPolicy retrieves a specific policy.
+	GetClusterPolicy(ctx kapi.Context, id string) (*authorizationapi.ClusterPolicy, error)
+	// CreateClusterPolicy creates a new policy.
+	CreateClusterPolicy(ctx kapi.Context, policy *authorizationapi.ClusterPolicy) error
+	// UpdateClusterPolicy updates a policy.
+	UpdateClusterPolicy(ctx kapi.Context, policy *authorizationapi.ClusterPolicy) error
+	// DeleteClusterPolicy deletes a policy.
+	DeleteClusterPolicy(ctx kapi.Context, id string) error
+}
+
+type WatchingRegistry interface {
```

Comment:
> why the separate type?

To keep the internal code coherent, `ClusterPolicy` has an adapting registry that can make the it appear as a `Policy.Registry` **except** for watches, which are really hard to adapt.  I turned out that most of the code that wanted to deal with both scope of `Policy` and didn't care which was which (all of the authorization code) didn't use the watch.  Splitting into a separate interface allowed for the proxying registry.

Given a proper "all" value for a namespace, there is no ambiguity about whether a request is for something without a namespace, something with a namespace, and something for all namespaces.  That made the behavior of the proxying registry consistent and would make a unified registry possible, (if we ever kill the three meanings of namespace="").